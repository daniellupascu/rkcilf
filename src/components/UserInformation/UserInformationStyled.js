import styled from "styled-components";

export const UserInfoContainer = styled.div`
  display: flex;
`;
