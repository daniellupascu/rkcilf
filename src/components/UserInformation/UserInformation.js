import React from "react";

import { UserInfoContainer } from "./UserInformationStyled";

const UserInformation = ({ user }) => {
  console.log({ user });
  return <UserInfoContainer>{user.id}</UserInfoContainer>;
};

export default UserInformation;
