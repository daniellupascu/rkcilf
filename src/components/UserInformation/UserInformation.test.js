import React from "react";
import renderer from "react-test-renderer";
import UserInformation from "./UserInformation";

describe("Test UserInformation Component", () => {
  it("renders correctly", () => {
    const user = {
      name: "Frank Johnson",
      html_url: "https://github.com/",
      blog: "https://github.com/",
      bio: "users bio",
      location: "Arizona",
      public_repos: 10,
      public_gists: 0,
      avatar_url:
        "https://github.githubassets.com/images/modules/logos_page/Octocat.png",
    };

    const tree = renderer.create(<UserInformation user={user} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("Test UserInformation Component with missing fields", () => {
    const user = {
      name: "Frank Johnson",
      html_url: "https://github.com/",
      location: "Arizona",
      public_repos: 10,
      public_gists: 0,
      avatar_url:
        "https://github.githubassets.com/images/modules/logos_page/Octocat.png",
    };

    const tree = renderer.create(<UserInformation user={user} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
