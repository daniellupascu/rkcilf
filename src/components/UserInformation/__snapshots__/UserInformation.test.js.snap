// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test UserInformation Component Test UserInformation Component with missing fields 1`] = `
<div
  className="UserInformationStyled__UserInfoContainer-sc-1blzddj-0 jenztY"
>
  <div
    className="sc-AxjAm jhiTqR"
  >
    <img
      className="sc-AxirZ fZpmTJ"
      src="https://github.githubassets.com/images/modules/logos_page/Octocat.png"
    />
  </div>
  <div
    className="UserInformationStyled__DetailsContainer-sc-1blzddj-1 kmhjpF"
  >
    <div
      className="UserInformationStyled__UserName-sc-1blzddj-2 cWlipe"
    >
      Frank Johnson
    </div>
    <div
      className="UserInformationStyled__UsernameContainer-sc-1blzddj-3 kVUCfQ"
    >
      <a
        className="UserInformationStyled__ProfileUrl-sc-1blzddj-4 eUbYqi"
        href="https://github.com/"
      />
    </div>
    <div
      className="UserInformationStyled__DetailTilesContainer-sc-1blzddj-6 cvHTUm"
    >
      <div
        className="sc-AxiKw lmEmTk"
      >
        <div
          className="sc-AxhCb bBcDsF"
        >
          Following
        </div>
        <div
          className="sc-AxhUy kMyssT"
        />
      </div>
      <div
        className="sc-AxiKw lmEmTk"
      >
        <div
          className="sc-AxhCb bBcDsF"
        >
          Followers
        </div>
        <div
          className="sc-AxhUy kMyssT"
        />
      </div>
    </div>
    <div
      className="UserInformationStyled__DetailTilesContainer-sc-1blzddj-6 cvHTUm"
    >
      <div
        className="sc-AxiKw lmEmTk"
      >
        <div
          className="sc-AxhCb bBcDsF"
        >
          Location
        </div>
        <div
          className="sc-AxhUy kMyssT"
        >
          Arizona
        </div>
      </div>
    </div>
    <div
      className="UserInformationStyled__DetailTilesContainer-sc-1blzddj-6 cvHTUm"
    >
      <div
        className="sc-AxiKw lmEmTk"
      >
        <div
          className="sc-AxhCb bBcDsF"
        >
          Repos
        </div>
        <div
          className="sc-AxhUy kMyssT"
        >
          10
        </div>
      </div>
      <div
        className="sc-AxiKw lmEmTk"
      >
        <div
          className="sc-AxhCb bBcDsF"
        >
          Gitsts
        </div>
        <div
          className="sc-AxhUy kMyssT"
        >
          0
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test UserInformation Component renders correctly 1`] = `
<div
  className="UserInformationStyled__UserInfoContainer-sc-1blzddj-0 jenztY"
>
  <div
    className="sc-AxjAm jhiTqR"
  >
    <img
      className="sc-AxirZ fZpmTJ"
      src="https://github.githubassets.com/images/modules/logos_page/Octocat.png"
    />
  </div>
  <div
    className="UserInformationStyled__DetailsContainer-sc-1blzddj-1 kmhjpF"
  >
    <div
      className="UserInformationStyled__UserName-sc-1blzddj-2 cWlipe"
    >
      Frank Johnson
    </div>
    <div
      className="UserInformationStyled__UsernameContainer-sc-1blzddj-3 kVUCfQ"
    >
      <a
        className="UserInformationStyled__ProfileUrl-sc-1blzddj-4 eUbYqi"
        href="https://github.com/"
      />
    </div>
    <div
      className="UserInformationStyled__BioContainer-sc-1blzddj-5 jdXPTn"
    >
      users bio
    </div>
    <div
      className="UserInformationStyled__DetailTilesContainer-sc-1blzddj-6 cvHTUm"
    >
      <div
        className="sc-AxiKw lmEmTk"
      >
        <div
          className="sc-AxhCb bBcDsF"
        >
          Following
        </div>
        <div
          className="sc-AxhUy kMyssT"
        />
      </div>
      <div
        className="sc-AxiKw lmEmTk"
      >
        <div
          className="sc-AxhCb bBcDsF"
        >
          Followers
        </div>
        <div
          className="sc-AxhUy kMyssT"
        />
      </div>
    </div>
    <div
      className="UserInformationStyled__DetailTilesContainer-sc-1blzddj-6 cvHTUm"
    >
      <div
        className="sc-AxiKw lmEmTk"
      >
        <div
          className="sc-AxhCb bBcDsF"
        >
          Location
        </div>
        <div
          className="sc-AxhUy kMyssT"
        >
          Arizona
        </div>
      </div>
      <div
        className="sc-AxiKw lmEmTk"
      >
        <div
          className="sc-AxhCb bBcDsF"
        >
          Blog
        </div>
        <div
          className="sc-AxhUy kMyssT"
        >
          <a
            className="UserInformationStyled__BlogUrl-sc-1blzddj-7 gkdhrO"
            href="https://github.com/"
          >
            https://github.com/
          </a>
        </div>
      </div>
    </div>
    <div
      className="UserInformationStyled__DetailTilesContainer-sc-1blzddj-6 cvHTUm"
    >
      <div
        className="sc-AxiKw lmEmTk"
      >
        <div
          className="sc-AxhCb bBcDsF"
        >
          Repos
        </div>
        <div
          className="sc-AxhUy kMyssT"
        >
          10
        </div>
      </div>
      <div
        className="sc-AxiKw lmEmTk"
      >
        <div
          className="sc-AxhCb bBcDsF"
        >
          Gitsts
        </div>
        <div
          className="sc-AxhUy kMyssT"
        >
          0
        </div>
      </div>
    </div>
  </div>
</div>
`;
